src/Microsoft.AspNet.Routing/RouteConstraintMatcher.cs

```diff
@@ -24,35 +22,30 @@ public static class RouteConstraintMatcher
                 return true;
             }
 
+            logger.LogVerbose("Verifying route constraints");
+            if(constraints != null && constraints.Count == 0)
+            {
+                logger.LogVerbose("No route constraints found.");
+            }
+
             foreach (var kvp in constraints)
             {
                 var constraint = kvp.Value;
+
+                logger.LogVerbose("Verifying constraint '{RouteConstraint}' for '{RouteKey}'", constraint, kvp.Key);
+
                 if (!constraint.Match(httpContext, route, kvp.Key, routeValues, routeDirection))
                 {
-                    if (routeDirection.Equals(RouteDirection.IncomingRequest)
-                        && logger.IsEnabled(LogLevel.Verbose))
+                    if (routeDirection.Equals(RouteDirection.IncomingRequest))
                     {
-                        logger.WriteValues(new RouteConstraintMatcherMatchValues()
-                        {
-                            ConstraintKey = kvp.Key,
-                            Constraint = kvp.Value,
-                            Matched = false
-                        });
+                        logger.LogWarning(
+                            "Route value for '{RouteKey}' did not match the constraint '{RouteConstraint}'.",
+                            kvp.Key,
+                            kvp.Value);
```

Comment:
This is not a warning, it's not really a mistake or a programming error, it should be verbose.